src/Role/Service/CreateChildRoleService.php

```diff
@@ -62,10 +63,24 @@ public function execute(array $data): CustomResponseInterface
             );
         }
 
-        /** @var int $parentChildrenCount */
-        $parentChildrenCount = count($this->roleHierarchyRepository->getChildIds([$parentRoleId]));
+        /** @var RoleHierarchy $lastChild */
+        $lastChild = $this->roleHierarchyRepository->getDataObjectManager()

```

Comment:
can you please outsource this into a custom function inside the given repository?